firefox-ios/Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -6,41 +6,36 @@ import Foundation
 import Shared
 import Common
 
-protocol ClipboardBarDisplayHandlerDelegate: AnyObject {
-    func shouldDisplay(clipBoardURL url: URL)
+@MainActor
+protocol ClipboardBarDisplayHandler {
+    var clipboardToast: ButtonToast? { get set }
+    func checkIfShouldDisplayBar()
+}
 
+protocol ClipboardBarDisplayHandlerDelegate: AnyObject {
+    @MainActor
     @available(iOS 16.0, *)
     func shouldDisplay()
 }
 
-class ClipboardBarDisplayHandler: NSObject {
-    public struct UX {
+@MainActor
+@available(iOS 16.0, *)
+final class DefaultClipboardBarDisplayHandler: ClipboardBarDisplayHandler {
```

Comment:
As you can see there's a bunch of logic in there that took me a while to understand. I don't see how we would be able to easily clean this up/detect those changes when we drop iOS 15, so that's why I took the time to create the legacy class. I think it's easier to understand what's happening now that way 🤞